test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
+                .withTasksDueThisWeekForProject(jest.fn(async () => done()))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withCommandController(commandControllerMock)
+                .withClient(client)
+                .build();
+                
+            const message = new MessageBuilder()
+                .withContent('!tasks due')
+                .withSend(jest.fn(() => Promise.resolve()))
+                .build();
+
+            client.emit('message', message);
+        });
+
+        it('should send warning message when channel type is not text' , () => {
+            const client = setupClient();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksDueThisWeekForProject(jest.fn(() => Promise.resolve(new RichEmbed())))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withCommandController(commandControllerMock)
+                .withClient(client)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks due')
+                .withChannelType('voice')
+                .build();
+
+            client.emit('message', message);
+
+            expect(message.channel.send)
+                .toBeCalledWith('!tasks due command must be called from a text channel' 
+                    + ' associated with a project');
+        });
+
+        it('should send error message and log when error getting project ID', done => {
+            expect.assertions(2);
+                
+            const client = setupClient();
+
+            const projectId = 0;
+            const error = 'error';
+            const channelName = 'channel';
+            let messagesSent = 0;
+
+            let sentMessageValue = '';
+
+            const loggerMock = new LoggerMockBuilder()
+                .withWarn(jest.fn().mockImplementation(value => {
+                    expect(value).toBe(`Error getting tasks due for week: Error: ${error}`);
+                    expect(sentMessageValue).toBe(`Unable to find ActiveCollab project for channel ` 
+                        + channelName);
+                    done();
+                }))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn(() => { throw Error(error); }))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withMappingController(mappingControllerMock)
+                .withClient(client)
+                .withLogger(loggerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks due')
+                .withSend(jest.fn(async value => {
+                    messagesSent++;
+
+                    if (messagesSent === 2) {
```

Comment:
Should we check the first message as well?

Reply:
There's other tests checking the first message, could do it here also but seemed unnecessary